src/components/Select.vue

```diff
@@ -0,0 +1,30 @@
+<template>
+  <select class="select" :name="name" @change="changeOption">
+    <option
+      v-for="(item, idx) in list"
+      :key="idx"
+      :value="item.code"
+      :selected="item.code === selectedOption"
+      class="select__option"
+    >
+      {{ item.language }}
+    </option>
+  </select>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from "vue-property-decorator";
+import { ILanguage } from "@/views/types";
+
+@Component
+export default class Select extends Vue {
+  @Prop({ default: [] }) list!: ILanguage[];
+  @Prop({ required: true }) name!: string;
+  @Prop({ required: true }) selectedOption!: string;
```

Comment:
I'm not sure about the `selectedOption` 
We pass `selectedOption` from a parent to a child as a prop. And on the other hand `Select` component can change the selected option and then pass it to the parent and parent will pass it back into the `Select` component.
I expected to see a `defaultOption`
Maybe `Select` component will have v-model `<Select v-model=.../>` and changing the selected option we can emit a change event to update the `v-model` in the parent component
Let me know if it makes sense for you and if you have doubts that the suggestion will work